gpt/commit.js

```diff
@@ -252,6 +252,61 @@ const runGenerativeCommit = async () => {
     }
```

Comment:
The tools are being reinitialized each time the recursiveAgent function is called. Consider initializing them once and passing them as arguments to the function to improve performance.